### PR TITLE
Add Component #26

### PR DESCRIPTION
**Pull request by Samaamro20**
- create add component 
 - [x] get lastKM from car model to car 
 - [x] calculate total trip to this log 
 - [x] add logs to logs model

### CODE REVIEW

File: client/src/App.js [hunk 1]
```diff
@@ -7,15 +7,58 @@ import Footer from "./components/Footer";
...
+import { addHelper } from "./utils/addHelper.js";
+
 import { BrowserRouter, Route } from "react-router-dom";
 import data from "./data.json";
 import "typeface-roboto";
 
 class App extends Component {
   state = {
-    carData: null
+    carData: 1,
```
Comment (m4v15) [inline client/src/App.js]: `carData` seems like a bad name for this `carId` instead?

File: client/src/App.js [hunk 2]
```diff
@@ -24,25 +67,41 @@ class App extends Component {
...
+            render={props => (
               <Cars {...props} data={data} handler={this.selectCar} />
             )}
           />
-          <Route exact path="/reports" component={Reports} />
-          <Route exact path="/" component={Login} />
-          <Route exact path="/" component={Home} />
+          <Route exact path="/home" component={Reports} />
+          <Route exact path="/login" component={Login} />
+          <Route exact path="/home" component={Home} />
```
Comment (m4v15) [inline client/src/App.js]: home has 2 matches

File: client/src/App.js [hunk 3]
```diff
@@ -7,15 +7,58 @@ import Footer from "./components/Footer";
...
+    purpose: "Personal",
+    driver_name: null,
+    start_km: null,
+    end_km: null,
+    total: null,
+    note: null,
+    failed:false
   };
 
+  getlastkm =()=>{
```
Comment (m4v15) [inline client/src/App.js]: Is this needed? I thought this was going to be stored in the car table anyway, so won't we already have this information from when we generate the car list?

File: client/src/App.js [hunk 4]
```diff
@@ -7,15 +7,58 @@ import Footer from "./components/Footer";
...
+    fetch("/getstartkm/"+this.state.carData).then(response=>{
+      return response.json()
+    }).then(result=>{
+      this.setState({start_km:result[0].last_log_km})
+    })
+  }
+
+  handleChange = name => event => {
+    console.log(event.target.value);
```
Comment (m4v15) [inline client/src/App.js]: remove console log

File: client/src/App.js [hunk 5]
```diff
@@ -7,15 +7,58 @@ import Footer from "./components/Footer";
...
+    this.setState({ [name]: event.target.value });
+    //console.log(this.state);
```
Comment (m4v15) [inline client/src/App.js]: remove

File: client/src/components/Add/asyncValidate.js [hunk 6]
```diff
@@ -0,0 +1,13 @@
+const sleep = ms => new Promise(resolve => setTimeout(resolve, ms))
```
Comment (m4v15) [inline client/src/components/Add/asyncValidate.js]: what does this file do?

Comment (m4v15) [reply client/src/components/Add/asyncValidate.js]: you should put an explanation in the PR description, QA shouldn't have to ask these things.

Would be easier if your issues were more clearly planned out...

File: src/database/models/index.js [hunk 7]
```diff
@@ -10,6 +10,7 @@ logs.belongsTo(car, {
   targetKey: 'id',
 });
 
+logs.hook('afterCreate', log => car.update({ last_log_km: log.end_km }, { where: { id: log.car_id } }));
```
Comment (m4v15) [inline src/database/models/index.js]: cooool

Comment (m4v15) [reply src/database/models/index.js]: I think I know what this does, can you clarify tho

File: client/src/App.js (hunk 1 above)
Comment (Samaamro20) [reply client/src/App.js]: Done

File: client/src/App.js (hunk 2 above)
Comment (Samaamro20) [reply client/src/App.js]: Done

File: client/src/App.js (hunk 4 above)
Comment (Samaamro20) [reply client/src/App.js]: Done

File: client/src/App.js (hunk 5 above)
Comment (Samaamro20) [reply client/src/App.js]: Done

File: client/src/components/Add/asyncValidate.js (hunk 6 above)
Comment (Samaamro20) [reply client/src/components/Add/asyncValidate.js]: its a unused file we remove it :+1:

File: src/database/models/index.js (hunk 7 above)
Comment (Samaamro20) [reply src/database/models/index.js]: As I said before https://github.com/FACK1/KamSafer/issues/43#issuecomment-464460700 :+1:

File: client/src/App.js (hunk 3 above)
Comment (Samaamro20) [reply client/src/App.js]: - because we will save a `car_id`  in next step , and we do not know if another user change it to same car .

**Pull request merged**